### PR TITLE
Update Azure messaging: settlement attributes and span names

### PR DESCRIPTION
**Pull request by lmolkova**
## Changes

- Adds Azure Service Bus-specific attribute describing settlement type (`complete`, `abandon`, `defer`, `dead_letter`)
- Clarifies operation names for settlement and deliver span

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* ~~schema-next.yaml~~

### CODE REVIEW

**Review by joaopgrassi — APPROVED** [review]

File: docs/messaging/azure-messaging.md [hunk 1]
```diff
@@ -12,13 +12,23 @@ The Semantic Conventions for [Azure Service Bus](https://learn.microsoft.com/azu
...
+- Spans for `create`, `receive`, and `publish` operations SHOULD follow general `<destination name> <operation name>` pattern
+
 ### Span attributes
 
 The following additional attributes are defined:
 <!-- semconv messaging.servicebus -->
 | Attribute  | Type | Description  | Examples  | Requirement Level |
 |---|---|---|---|---|
 | [`messaging.servicebus.destination.subscription_name`](../attributes-registry/messaging.md) | string | The name of the subscription in the topic messages are received from. | `mySubscription` | Conditionally Required: If messages are received from the subscription. |
+| [`messaging.servicebus.disposition_status`](../attributes-registry/messaging.md) | string | Describes [settlement type](https://learn.microsoft.com/azure/service-bus-messaging/message-transfers-locks-settlement#peeklock). | `complete` | Conditionally Required: if and only if `messaging.operation` is `settle`. |
```
Comment (pyohannes) [inline docs/messaging/azure-messaging.md:31]:
```suggestion
| [`messaging.servicebus.disposition_status`](../attributes-registry/messaging.md) | string | Describes the [settlement type](https://learn.microsoft.com/azure/service-bus-messaging/message-transfers-locks-settlement#peeklock). | `complete` | Conditionally Required: if and only if `messaging.operation` is `settle`. |
```

**Review by pyohannes — APPROVED** [review]
Looks good, just some nit-picks.

File: docs/messaging/azure-messaging.md [hunk 2]
```diff
@@ -29,6 +39,15 @@ The following additional attributes are defined:
...
 `messaging.system` MUST be set to `"eventhubs"`.
 
+### Span names
+
+The span name SHOULD follow [general messaging span name pattern](../messaging/azure-messaging.md): it SHOULD start with the messaging destination name (Event Hubs namespace) and
+contain a low-cardinality name of an operation span describes:
+
+- Spans for `settle` operation SHOULD follow  `<destination name> checkpoint` (matching Event Hubs terminology)
+- Spans names for `publish` operation SHOULD follow `<destination name> send` pattern
+- Spans for `create`, `receive`, and `publish` operations SHOULD follow general `<destination name> <operation name>` pattern
```
[the remark this answers is not in the document]
Comment (lmolkova) [reply docs/messaging/azure-messaging.md]: Thank you!

**Comment by joaopgrassi** [comment]
@lmolkova  Do you maybe want to set the stability on these before we merge?

**Review by joaopgrassi — APPROVED** [review]

**Pull request merged**